packages/terra-disclosure-manager/src/DisclosureManager.jsx

```diff
@@ -286,6 +287,7 @@ class DisclosureManager extends React.Component {
       disclosureSize: size,
       disclosureDimensions: dimensions,
       disclosureComponentKeys: [data.content.key],
+      disclosureTypeConfig: data.typeConfig,
```

Comment:
Should this be scoped to the disclosureComponentData[key]?

Reply:
I dont think so. 
CC: @tbiethman

Reply:
This name is fine, it aligns with the disclosure components type config prop: https://github.com/cerner/terra-framework/blob/4d5bef74cd7f17b16468058985b2ef1b0703de7a/packages/terra-disclosure-manager/src/DisclosureManager.jsx#L536

Reply:
@emilyrohrbough @lokesh-0813 it all depends on whether we want to handle this type config for the _first_ disclosure only (like we do with size) or for each subsequent disclosure as well. 

It would definitely feel a little weird to have the panel transitioning from squish to overlay and back again through the course of progressive disclosure in the panel. But we will definitely run into scenarios where teams say "I said squish, why isn't is squishing?" like they do with the size today.

I say let's leave this like it is and only support changing the type on the first disclosure. We can revisit if we have to later.